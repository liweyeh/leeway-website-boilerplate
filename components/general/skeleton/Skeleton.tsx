/**
 * Skeleton component
 * This serves as placeholder whilst the data is loading
 * @author Carlos LÃ³pez Rodriguez <clopez@adasasistemas.com>
 */

import styles from './skeleton.module.scss';

interface SkeletonProps {
	customStyles?: string;
}

export const Skeleton = ({ customStyles }: SkeletonProps) => {
	return <div className={[styles.skeleton, customStyles].join(' ')} />;
};
