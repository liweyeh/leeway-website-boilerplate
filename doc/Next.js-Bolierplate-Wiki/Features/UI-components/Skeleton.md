[[_TOC_]]

# Overview

- Purpose: A placeholder UI when data is loading
- Type: General UI
- Status: ðŸš€ **Completed**
- Code: [URL](https://dev.azure.com/ADASA-Accelerator/Website-React-Booster/_git/website-boilerplate-nextjs?path=/components/general/skeleton)
- Created by: Liwei Yeh <liwei.yeh@adasasistemas.com>
- Created time: 25/08/2022

# Props

- customStyles<`string`>: custom styling of the skeleton

# Usage

```jsx
<Skeleton />
```
