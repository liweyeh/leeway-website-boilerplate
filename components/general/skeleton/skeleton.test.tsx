import { render } from '@testing-library/react';
import { Skeleton } from './Skeleton';
import '@testing-library/jest-dom';

const setup = () => {
	const { asFragment } = render(<Skeleton />);
	return { asFragment };
};

test('test snapshot', () => {
	const { asFragment } = setup();
	expect(asFragment()).toMatchSnapshot();
});
