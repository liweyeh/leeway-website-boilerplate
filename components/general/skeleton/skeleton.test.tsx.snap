// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<DocumentFragment>
  <div
    class="skeleton "
  />
</DocumentFragment>
`;
